### PR TITLE
Pagination component

### PR DESCRIPTION
**Pull request by sidra2022**

### CODE REVIEW

**Comment by louislingatong** [comment]
Nice work! 

Please merge it to the latest code before merging to main.

File: packages/stencil/src/components/datacom-pagination/datacom-pagination.css [hunk 1]
```diff
@@ -0,0 +1,127 @@
+@import url("../../global/global.css");
```
Comment (louislingatong) [inline packages/stencil/src/components/datacom-pagination/datacom-pagination.css:1]: I think there is no need to import global.css here.

File: packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  @Listen('input', { capture: true })
+  onInput(event: InputEvent): void {
+    const inputElement = event.target as HTMLInputElement;
+    const newValue =
+      inputElement.value !== '' ? +inputElement.value : undefined;
+    this.currentPage = newValue;
+    this.pageChanged.emit(this.currentPage);
+  }
+
+  handleInputBlur = (event: Event): void => {
```
Comment (louislingatong) [inline packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx:87]: Please change the parameter typing from `Event` to `FocusEvent`

File: packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+  onClickLastPageHandler = () => {
+    this.currentPage = this.totalPages;
+    this.pageChanged.emit(this.currentPage);
+  };
+
+  /**
+   * Update the value when enter the page number in the input field.
+   */
+  onPageChangeHandler = (event: Event): void => {
```
Comment (louislingatong) [inline packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx:68]: Please change the parameter typing from `Event` to `KeyboardEvent` | `MouseEvent`

**Review by louislingatong — CHANGES_REQUESTED** [review]
Added tiny comments on the PRs. Thanks.

File: packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  @Listen('input', { capture: true })
+  onInput(event: InputEvent): void {
+    const inputElement = event.target as HTMLInputElement;
+    const newValue =
+      inputElement.value !== '' ? +inputElement.value : undefined;
+    this.currentPage = newValue;
+    this.pageChanged.emit(this.currentPage);
+  }
+
+  handleInputBlur = (event: FocusEvent): void => {
```
Comment (louislingatong) [inline packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx:88]: Please make this method to private. Thanks.

File: packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  private onClickLastPageHandler = () => {
+    this.currentPage = this.totalPages;
+    this.pageChanged.emit(this.currentPage);
+  };
+
+  /**
+   * Update the value when enter the page number in the input field.
+   */
+  @Watch('currentPage')
+  WatchChange(): void {
```
Comment (louislingatong) [inline packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx:69]: Please update the method name start with small letter and in camelCase format. Thanks.

**Review by louislingatong — COMMENTED** [review]
Looks good to me. though, added minor comments and it's good to merge to the main branch.

File: packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx (hunk 4 above)
Comment (sidra2022) [reply packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx]: Done

File: packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx (hunk 5 above)
Comment (sidra2022) [reply packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx]: Done

**Review by louislingatong — APPROVED** [review]
Perfect. Good job!

**Pull request merged**